lib/p5.sound.js

```diff
@@ -6771,11 +6769,20 @@ effect = function () {
     this.output.disconnect();
     this.output = undefined;
   };
+  p5.Effect.prototype.dispose = function () {
+    var index = p5sound.soundArray.indexOf(this);
+    p5sound.soundArray.splice(index, 1);
+    this.input.disconnect();
+    this.input = undefined;
+    this.output.disconnect();
+    this.output = undefined;
```

Comment:
* Assigning _undefined_ or `null`, and even assigning anything else to the property, is fully enough for GC purposes.
* Operator `delete` is for removing the property itself from its object as well.
* If we execute `this.input = undefined;` or `this.input = null;`, and then `console.log('input' in this);`, it's still gonna print `true`!
* Only for when we `delete this.input;`, `console.log('input' in this);` is gonna print `false`.
* Therefore for 100% purge, use operator `delete` rather than simply assinging _undefined_ or `null` to the property.  ;-)